pkg/processor/build/builder.go

```diff
@@ -324,9 +331,30 @@ func (b *Builder) resolveFunctionPath(functionPath string) (string, error) {
 		return "", fmt.Errorf("Function path doesn't exist: %s", resolvedPath)
 	}
 
+	if util.IsCompressed(resolvedPath) {
+		resolvedPath, err = b.decompressFunctionArchive(resolvedPath)
+		if err != nil {
+			return "", errors.Wrap(err, "Failed to decompress function archive")
+		}
+	}
+
 	return resolvedPath, nil
 }
 
+func (b *Builder) decompressFunctionArchive(functionPath string) (string, error) {
+	// create a staging directory
+	tempDir, err := ioutil.TempDir("", "nuclio-decompress-")
```

Comment:
Is this usage coherent with how we create temporary directories in other places in the code? Is it cleaned up later?

Reply:
For now let's leave without cleanup - there is a future task to clean up any temp dirs created during build.